subsys/esb/esb.c

```diff
@@ -1745,6 +1762,13 @@ static void radio_irq_handler(void)
 		}
 	}
 #endif /* defined(CONFIG_ESB_FAST_CHANNEL_SWITCHING) */
+
+	if (nrf_radio_int_enable_check(NRF_RADIO, NRF_RADIO_INT_END_MASK) &&
```

Comment:
This section inside the `if` block will not be executed on nRF52 and nRF53 devices where `ESB_RADIO_EVENT_END` equals 'NRF_RADIO_EVENT_END' and the event will be cleared by the `if` block in line 1744.